### PR TITLE
[WIP] Added Priority Queue

### PR DESCRIPTION
**Pull request by sarthakforwet**
#### References to other Issues or PRs or Relevant literature
Refrenced Issue #131 
 

#### Other comments
The following change to `queue.py` was done and it is updated with priority queue functionaliity using python `list` object . I am also including Priority Queues Implemented via Binary Heaps and that work is in progress .

### CODE REVIEW

File: pydatastructs/miscellaneous_data_structures/queue.py [hunk 1]
```diff
@@ -122,3 +127,50 @@ def __str__(self):
...
+        return ''.join([str(i) for i in self.queue])
+
+    def insert(self,data):
+        self.queue.append(data)
+
+    def deleteHighestPriority(self,priority="max"):
+        if self.is_empty:
+            raise ValueError("Queue is Empty")
+        if priority=="max":
+            max_ = max(self.queue)
```
Comment (HarsheetKakar) [inline pydatastructs/miscellaneous_data_structures/queue.py]: cant we use heap to do this? checking the max every time deleteHighestPriority is called seems to be an O(n) job using a heap can reduce it to O(log(n))

Comment (sarthakforwet) [reply pydatastructs/miscellaneous_data_structures/queue.py]: Yes as I have mentioned in my PR i am working on Binary Heaps for Priority Queue . This is one of the naive approaches for Priority Queue .

Comment (sarthakforwet) [reply pydatastructs/miscellaneous_data_structures/queue.py]: Yes as I have mentioned in my PR i am working on Binary Heaps for Priority Queue . This is one of the naive approaches for Priority Queue .

**Comment by czgdp1807** [comment]
I think `PriorityQueue` can have a framework similar to `Stack` i.e., a base class acting as an interface and it's subclasses having specific implementations i.e., `BinaryHeapPriorityQueue`, `FibonacciHeapPriorityQueue`, etc.

**Comment by sarthakforwet** [comment]
@czgdp1807 sure implementation is in progress .

**Comment by sarthakforwet** [comment]
@czgdp1807 I've added Priority Queue as you mentioned like the Stack manner using `Binary Heap` .
Now I am implementing Priority Queue using `Fibbonacci Heap`

**Comment by czgdp1807** [comment]
Closing in reference to https://github.com/codezonediitj/pydatastructs/wiki/Pull-Request-Policy#when-to-close-a-pr

**Pull request closed without merging**